### PR TITLE
OkHttpJsonApi#getMediaList migrated to retrofit

### PR DESCRIPTION
**Pull request by ilgazer**
This PR migrates OkHttpJsonApi#getMediaList to 2 different functions, MediaClient#getMediaListFromSearch and MEdiaClient#getMediaListFromCategory.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/category/CategoryImageController.java [hunk 1]
```diff
@@ -25,6 +25,6 @@ public CategoryImageController(OkHttpJsonApiClient okHttpJsonApiClient) {
      * @return
      */
     public Single<List<Media>> getCategoryImages(String categoryName) {
-        return okHttpJsonApiClient.getMediaList("category", categoryName);
+        return null;// okHttpJsonApiClient.getMediaList("category", categoryName);
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryImageController.java]: @ilgazer Can you update this. :)

**Comment by ilgazer** [comment]
That class is no longer used and will be deleted. Thanks for catching that.

6 Temmuz 2019 Cumartesi tarihinde Vivek Maskara <notifications@github.com>
yazdı:

File: app/src/main/java/fr/free/nrw/commons/category/CategoryImageController.java (hunk 1 above)
Comment (ilgazer) [reply app/src/main/java/fr/free/nrw/commons/category/CategoryImageController.java]: Done :)

**Comment by ilgazer** [comment]
> The issue is that the category passed is not suffixed with "Category:".

Maybe we should set a standart on whether we should pass around page names with namespace or not throughout the app.

**Pull request merged**